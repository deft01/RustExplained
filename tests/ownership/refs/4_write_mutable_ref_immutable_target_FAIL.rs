#![allow(warnings)]

fn main() {
    let target: usize = 0;

    let ref_foo = &mut target;
    let ref_bar = &mut target;
    let ref_baz = &mut target;

    println!("{}", ref_baz);
}
