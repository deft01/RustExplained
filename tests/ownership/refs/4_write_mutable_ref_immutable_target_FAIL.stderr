error[E0596]: cannot borrow `target` as mutable, as it is not declared as mutable
 --> tests/ownership/refs/4_write_mutable_ref_immutable_target_FAIL.rs:4:9
  |
4 |     let target: usize = 0;
  |         ^^^^^^ not mutable
5 |
6 |     let ref_foo = &mut target;
  |                   ----------- cannot borrow as mutable
7 |     let ref_bar = &mut target;
  |                   ----------- cannot borrow as mutable
8 |     let ref_baz = &mut target;
  |                   ----------- cannot borrow as mutable
  |
help: consider changing this to be mutable
  |
4 |     let mut target: usize = 0;
  |         +++
